Pocket/pocket-ios
# Add Universal Links support

## Summary
*  This PR adds support for universal links, and add routes for the following types
    - Collections
    - Syndicated articles

## Implementation Details
Major changes
* Add `Route`, a protocol whose concrete implementations define routes for universal links
* Add `LinkRouter` a configurable router that accepts a number of routes and matches an url against them
* Refactor `RecommendationViewModel` to `RecommendableViewModel`, to handle externally linked syndicated articles

## Test Steps
> [!IMPORTANT]  
> To properly test this PR, because we haven't deployed an updated version of the `apple-app-site-association` file, you need to setup a local testing environment for it. 
> this requires the following steps:
> - obtain the updated file
> - Configure your mobile device for deep link development 
> - redirect the device to obtain the updated file from a local https server (e.g. install Proxyman and define a local mapping)
> for more details on how to setup the test environment you can reach out to me or @bassrock 


* Once the setup is complete, build/run this branch
* obtain external links to collections and/or syndicated articles, e. g.

https://getpocket.com/explore/item/why-exhaustion-is-not-unique-to-our-overstimulated-age

* from your device, on an external app, tap the link and make sure it opens Pocket and shows the linked article in the corresponding native view (collection or syndicated article)

## PR Checklist:
